pinot/net.py

```diff
@@ -18,22 +18,78 @@ class Net(torch.nn.Module):
     learning, puts into a predicted distribution and calculates the
     corresponding divergence.
 
+
+    Attributes
+    ----------
+    representation: a `pinot.representation` module
+        the model that translates graphs to latent representations
+    output_regression: a `torch.nn.Moudle` or None,
+        if None, this will be set as a simple `Linear` layer that inputs
+        the latent dimension and output the number of parameters for
+        `self.distribution_class`
+    distribution_class: a `torch.distribution` object,
```

Comment:
Do we need both this and the noise model? distribution class plays the role of the noise model here. What is your reasoning for this redundancy?

Reply:
And to clarify: I would prefer to only pass noise_model and subject to the choice of noise model construct the right likelihood later on in one of the forward-type functions.

Why have all this duplication?

Reply:
I was trying to allow manipulation of sigma.

Reply:
what do you mean?